### ISSUE TITLE
User Interface to edit WATM CSV file

### ISSUE THREAD

**Issue by asirota** [NewPath-Consulting/watm#36]
Requirements

- end user ability to add a new WATM configuration without code
- update an existing configuration
- delete an existing configuration
- ability to define the following properties for any CLASS/ID - font, font-size, color, background-color, visibility, hover pseudo classes, mouse in/out
- certain properties would be available only for certain HTML entities (button, div, link, form, a, etc)
- for text changes ability to introduce alternate text or translation text
- download a CSV file for reuse
[
Have a look at a starting point CSV here](https://www.dropbox.com/sh/242lqcr0w4b5m84/AADZeFBYYLRcLqED_eXqzsbla?dl=0)

**Comment by mDisna** [comment]
I have two proposals for this, and have linked to their respective design documents. The first is easier to implement, the second is more full-featured.


**[Element Inspector - Download configuration](https://docs.google.com/document/d/1Wd2HyKHV4d7yacL47KVrOQM6bWrs-Wapnsk7pLqQm4g/edit?usp=sharing)**
This solution solution will add a container to the bottom of websites where you can set the configuration for selected elements. The configuration can then be copy/pasted or downloaded to update the WATM CSV file on the client's website. Existing configurations would not be taken into account.

**[Remote Editor - Save and Load Configuration Remotely](https://docs.google.com/document/d/1zkCoT44i6ECtczRZ5OE_XrmbdClePQYGF1XdaavKCEk/edit?usp=sharing)**
This solution would require us to store the configuration, eliminating the need to upload a CSV file to the client's website. The interface would let you select elements on the client's website (loaded in an iframe) and let you view any existing configuration for that element, which you can modify, delete, or add to.

**Comment by asirota** [comment]
This library may be useful for downloading files https://github.com/rndme/download

**Comment by mDisna** [comment]
I have done some research in to webdav through JS. None of the examples I found online worked, but I was able to modify one to get it working.

Some things I found:
* Webdav through JS can only access files on the same domain
* The JS can be loaded remotely
* Submitting the authentication via code does not appear to be working, but users will be prompted with a login box on first load

In my test I was able to connect to WA webdav, list the files, read a file, and write to a file.

I have created a new design document - [visible here](https://docs.google.com/document/d/1_ChlXpy8wyyqqo0Xn07qKOFYXiJqIwG48SDJiGh_cKg/edit?usp=sharing)
